Assets/Scripts/CombatManager.cs

```diff
@@ -23,6 +27,9 @@ public class CombatManager : MonoBehaviour {
             hp.SetupCombat();
         }
 
+        _audio = this.GetComponent<AudioSource>();

```

Comment:
I have two best practices I've been trying  so far to lessen risk of runtime null pointer exceptions. For example, if we deleted the AudioSource in the future, this could would still compile, but it'd fail at runtime in the next line, right? 

I'd love if more things could be caught compile time, but not sure if possible. Thoughts?

1. Create the link explicitly in the editor (make it `public AudioSource _audio` above)

![image](https://user-images.githubusercontent.com/102242/82990387-f7486a00-9fb0-11ea-8ab8-0a0c98545deb.png)

=> you can still get a runtime null pointer exception if the thing isn't defined in the editor. It's also possible to override what you define in the editor. 

2. **fail fast.** when I'm doing my setup in Start/Awake, assert that the object exists. 

=> It's verbose, but it'll catch errors quickly.

https://github.com/CleverGameDev/Cleverfall/blob/3b7322b55a18129a515b06128723be6eb9bc10a1/Assets/Scripts/Wraparound.cs#L11-L12

--

Reply:
Gotcha! Seems weird to me to make a public variable for a component on the object, but I can do something similar to the second suggestion. 
Probably will end up doing something like [this](https://docs.unity3d.com/ScriptReference/RequireComponent.html). That allows the Unity editor to automatically check for this.

Reply:
Excellent! I like your approach better, thank for sharing :D